modules/java_api/src/main/cpp/core.cpp

```diff
@@ -76,6 +76,73 @@ JNIEXPORT jlong JNICALL Java_org_intel_openvino_Core_CompileModel(JNIEnv *env, j
     return 0;
 }
 
+JNIEXPORT jlong JNICALL Java_org_intel_openvino_Core_CompileModel1(JNIEnv *env, jobject obj, jlong coreAddr, jstring path)
+{
+    JNI_METHOD("CompileModel1",
+        std::string n_path = jstringToString(env, path);
+
+        Core *core = (Core *)coreAddr;
+        CompiledModel *compiled_model = new CompiledModel();
+        *compiled_model = core->compile_model(n_path);
+
+        return (jlong)compiled_model;
+    )
+    return 0;
+}
+
+JNIEXPORT jlong JNICALL Java_org_intel_openvino_Core_CompileModel2(JNIEnv *env, jobject obj, jlong coreAddr, jstring path, jstring device)
+{
+    JNI_METHOD("CompileModel2",
+        std::string n_device = jstringToString(env, device);
+        std::string n_path = jstringToString(env, path);
+
+        Core *core = (Core *)coreAddr;
+        CompiledModel *compiled_model = new CompiledModel();
+        *compiled_model = core->compile_model(n_path, n_device);
+
+        return (jlong)compiled_model;
+    )
+    return 0;
+}
+
+JNIEXPORT jlong JNICALL Java_org_intel_openvino_Core_CompileModel3(JNIEnv *env, jobject obj, jlong coreAddr, jstring path, jstring device, jobject props)
+{
+    JNI_METHOD("CompileModel3",
+        std::string n_device = jstringToString(env, device);
+        std::string n_path = jstringToString(env, path);
+        AnyMap map;
+        for (const auto& it : javaMapToMap(env, props)) {
+            map[it.first] = it.second;
+        }
+
+        Core *core = (Core *)coreAddr;
+        CompiledModel *compiled_model = new CompiledModel();
```

Comment:
Just for understanding in future :)
```suggestion
        // AnyMap will be copied inside compile_model, so we don't have to track this object lifetime
        CompiledModel *compiled_model = new CompiledModel();
```